### PR TITLE
Add create student endpoint

### PR DESCRIPTION
**Pull request by femiayodeji**
Task one - create student

### CODE REVIEW

File: Controllers/StudentsController.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            _repository = repository;
+        }
+
+        //POST api/students
+        [HttpPost]
+        public ActionResult <Student> CreateStudent(Student student){
+            if(student == null){
+                return BadRequest();
+            }
+            _repository.CreateStudent(student);
```
Comment (johnojetunde) [inline Controllers/StudentsController.cs]: You shouldn't have repository directly in the controller

Comment (femiayodeji) [reply Controllers/StudentsController.cs]: How better could I have done it?

Comment (femiayodeji) [reply Controllers/StudentsController.cs]: I have added service layer for the application logic.

File: Controllers/StudentsController.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+            _studentService = studentService;
+        }
+
+        //POST api/students
+        [HttpPost]
+        public ActionResult <Student> CreateStudent(Student student){
+            if(student == null){
+                return BadRequest();
+            }
+            _studentService.CreateStudent(student);
```
Comment (johnojetunde) [inline Controllers/StudentsController.cs:24]: the implementation of the createStudent method in the service should determine how to save the student;
The saving of the student shouldn't be exposed to the service.
Do you understand the concept of low coupling and high cohesion in Software

**Review by johnojetunde — CHANGES_REQUESTED** [review]

File: Controllers/StudentsController.cs (hunk 2 above)
Comment (femiayodeji) [reply Controllers/StudentsController.cs:24]: I do but I prolly need to study them more.

Comment (femiayodeji) [reply Controllers/StudentsController.cs:24]: I have loosened the coupling and maintained the inheritance for cohesion.

**Pull request merged**